app/src/main/kotlin/viewmodel/graph/EdgeViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package viewmodel.graph
+
+import androidx.compose.runtime.State
+import androidx.compose.ui.unit.Dp
+import ARROW_DEPTH
+import ARROW_SIZE
+import SQRT_3
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.graphics.Color
+import kotlin.math.sqrt
+
+class EdgeViewModel<D>(
+    val firstVertex: VertexViewModel<D>,
+    val secondVertex: VertexViewModel<D>,
+    private val isDirected: State<Boolean>
+) {
+
+    fun isDirected() = isDirected.value
+
+    private val radius = firstVertex.radius
+
+    var highlightColor = mutableStateOf(Color.LightGray)
+
+    internal fun calculateFirstVertexCenter(scale: Float): Pair<Dp, Dp> {
```

Comment:
Is there any benefit of this function being public instead of internal?

Reply:
`public` is the default so it is the reason for using `internal` that needs to be explained :)

`internal` hides the function from outer [modules](https://kotlinlang.org/docs/visibility-modifiers.html#modules) but you only have a single one in your project (besides `test` which is a special case that is not influenced by `internal` anyway) so it is effectively equivalent to `public`. It would make sense to use `internal` if you were developing a library: in that case the function would be available everywhere in your project but not for the users of the library, e.g. if they imported it from Maven via Gradle.